Source/SwiftLintFramework/Rules/JoinedDefaultRule.swift

```diff
@@ -54,11 +58,66 @@ public struct JoinedDefaultParameterRule: ASTRule, ConfigurationProviderRule, Op
             let argumentBodyOffset = argument.bodyOffset,
             let argumentBodyLength = argument.bodyLength,
             argument.name == "separator"
-            else {
-                return nil
-        }
+            else { return nil }
 
         let body = file.contents.bridge().substringWithByteRange(start: argumentBodyOffset, length: argumentBodyLength)
         return body == "\"\"" ? argumentBodyOffset : nil
     }
+
+    // MARK: - CorrectableRule
+
+    public func correct(file: File) -> [Correction] {
+        let violatingRanges = violationRanges(in: file, dictionary: file.structure.dictionary)
+        let matches = file.ruleEnabled(violatingRanges: violatingRanges, for: self)
+        var correctedContents = file.contents
+        var adjustedLocations: [Int] = []
+
+        for violatingRange in matches.reversed() {
+            if let range = file.contents.nsrangeToIndexRange(violatingRange) {
+                correctedContents = correctedContents.replacingCharacters(in: range, with: "")
+                adjustedLocations.insert(violatingRange.location, at: 0)
+            }
+        }
+
+        file.write(correctedContents)
+
+        return adjustedLocations.map {
+            Correction(ruleDescription: type(of: self).description, location: Location(file: file, characterOffset: $0))
+        }
+    }
+
+    private func violationRanges(in file: File, dictionary: [String: SourceKitRepresentable]) -> [NSRange] {
+        return dictionary.substructure.flatMap { subDictionary -> [NSRange] in
```

Comment:
Please avoid duplicating the logic for violation ranges/offsets.

You can check `EmptyParenthesesWithTrailingClosureRule.swift` as an example.